### ISSUE TITLE
Selecting all participants in booking

### ISSUE THREAD

**Issue by gemguardian** [Wunderbyte-GmbH/moodle-mod_booking#81]
Hello David,
It's not part of the excisting feature, and I have know idea how much time/cost it would be to add this feature. 

The scenario: 
The standard setting is 25 n records, so when I have more then 25 bookings on an option it shows page 1 of 2. If I selected the booked users to send (by example) a custom message, it doesn't select all. 

Isn't there away instead of changing the number of records to have an option simalar to the Moodle options at "people" where you can say "show all".. 

Thank you 

Gemma

**Comment by dasistwas** [comment]
Hi Gemma,

you can always change the number per page in the booking settings in "Miscellaneous settings->N. of records - pagination". Does that solve your problem? There would be a possibility to select the number of records from a dropdown menu, that should be shown directly on the page when viewing booked users, but that would require programming this special feature.

Kind regards,
David

**Comment by gemguardian** [comment]
I actually tried that, it does show the full list.
But when I select all, and click on Send a custom messages.

I get a bug error 

 









 

Van: David Bogner [mailto:notifications@github.com] 
Verzonden: zaterdag 11 maart 2017 14:56
Aan: dasistwas/moodle-mod_booking <moodle-mod_booking@noreply.github.com>
CC: gemguardian <gemma@ltnc.nl>; Author <author@noreply.github.com>
Onderwerp: Re: [dasistwas/moodle-mod_booking] Selecting all participants in booking (#81)

 

Hi Gem,

you can always change the number per page in the booking settings in "Miscellaneous settings->N. of records - pagination". Does that solve your problem? There would be a possibility to select the number of records from a dropdown menu, that should be shown directly on the page when viewing booked users, but that would require programming this special feature.

Kind regards,
David

**Comment by gemguardian** [comment]
I noticed the email didn't send the bug error. Enclosed the screenshot. 
![1](https://cloud.githubusercontent.com/assets/16559999/23824615/e0a8408c-0679-11e7-8d89-130964a74b70.jpg)

**Comment by dasistwas** [comment]
Hi Gemma,

thank you for the bug report. I can not reproduce this on my testing system. Which version do you use? What do you do exactly in order to get this error? Select all users on the "Manage bookings" page and then click on "Send custom message"? When I do this, the URL looks like that:
sendmessage.php?id=102&optionid=1&uids=a%3A3%3A{i%3A0%3Bi%3A5619%3Bi%3A1%3Bi%3A5709%3Bi%3A2%3Bi%3A5758%3B}

The parameter uids is present in the URL, so the error "missing param" should not happen. 

Kind regards,
David

**Comment by gemguardian** [comment]
hi David, 
I realized I didn't update the plug-in yet to your latest version. (right now it's version 2017021700/moodle 3.0)
I will ask the technical team to do this first, just to see if it solves the bug. 

Get back to you as soon as possible. 

Gemma

**Comment by gemguardian** [comment]
Hi David, 

I just tried again - after updates same error. 

Moodle version: 3.0.9
Booking version: 2017030600 

The booking option had already registrations when I changed the setting  **N. of records - pagination** from 25 to 50. 

I click in the booking option on "manage x responses"
I check all booked users
Click on Send custom Message

What is most stange is that I only receive the error on the booking options that are more then 40, on the booking option that says 28, I don't receive this error. 

Gemma
